### PR TITLE
interface for lexicon

### PR DESCRIPTION
**Pull request by mullikine**
No longer experiencing the lag I had.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -193,58 +183,30 @@ func formatSeverity() string {
...
-		warning("%s", err)
-		status |= 2
+	if err != nil {
+		return nil, err
 	}
-	elapsed := time.Since(start)
-	debug("total elapsed time %s", elapsed)
-	os.Exit(status)
+
+	issues, _ := runLinters(linters, paths, config.Concurrency, exclude, include)
```
Comment (BlakeMScurr) [inline main.go]: We should capture this error output.

File: main.go [hunk 2]
```diff
@@ -193,58 +183,30 @@ func formatSeverity() string {
...
+
+	return outputToIssues(issues), nil
```
Comment (BlakeMScurr) [inline main.go]: Returning a chan of issues is fine. We stream nodes anyway.

File: main.go [hunk 3]
```diff
@@ -336,10 +307,12 @@ func resolvePaths(paths, skip []string) []string {
 					return filepath.SkipDir
 				case !i.IsDir() && !skip && strings.HasSuffix(p, ".go"):
 					dirs.add(filepath.Clean(filepath.Dir(p)))
+					// dirs.add(filepath.Clean(p)) // DISCARD Make it allow file paths too. see below
```
Comment (BlakeMScurr) [inline main.go]: Why do we need filepaths?

Comment (mullikine) [reply main.go]: I have removed the comment here. Support was found for filepaths below that block of code .

**Review by emersonwood — APPROVED** [review]

**Pull request merged**